### PR TITLE
No scanf tokenstream

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

**Comment by mgeplf** [comment]
It passes all the tests, including the ones from NeuroM.  Please let me know if it's understandable.

**Comment by mgeplf** [comment]
As of 081aeca8be7db283338d6cc55b35fa06ae6d2fba this improves performance (parsing an SWC goes from 1.01ms to 0.817ms) and supports locales.

File: include/morphio/errorMessages.h [hunk 1]
```diff
@@ -153,12 +152,21 @@ class ErrorMessages
     /** Section self parent error message */
     std::string ERROR_SELF_PARENT(const Sample& sample) const;
 
+    /** The end of the file was reached before parsing finshed */
```
Comment (eleftherioszisis) [inline include/morphio/errorMessages.h:155]:
```suggestion
    /** The end of the file was reached before parsing finished */
```

File: src/readers/utils.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#include "./utils.h"
+
+namespace morphio {
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
+#define freelocale _free_locale
+#define strtol_l _strtol_l
```
Comment (eleftherioszisis) [inline src/readers/utils.cpp:6]: Maybe add a comment here?

File: src/readers/utils.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#endif
+
+StringToNumber::StringToNumber()
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
+    : locale(_create_locale(LC_ALL, "C")) {
+}
+#else
+    : locale(newlocale(LC_NUMERIC_MASK, "POSIX", nullptr)) {
+}
+#endif
```
Comment (eleftherioszisis) [inline src/readers/utils.cpp:31]: Same here. It's not clear to me why these are configured this way.

**Pull request closed without merging**